crank/bin/mcd_calculate.py

```diff
@@ -0,0 +1,166 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# mcd_calculate.py
+# Copyright (C) 2020 Wen-Chin HUANG
+#
+# Distributed under terms of the MIT license.
+#
+
+import json
+import os
+import sys
+import argparse
+import logging
+import multiprocessing as mp
+import numpy as np
+import pysptk
+import pyworld as pw
+import scipy
+from fastdtw import fastdtw
+
+from pathlib import Path
+import soundfile as sf
+from parallel_wavegan.utils import find_files
+from sprocket.speech import FeatureExtractor
+from crank.net.trainer.dataset import read_feature
+from crank.utils import load_yaml, open_featsscp
+from crank.utils import low_cut_filter
+from crank.feature import Feature
+
+def get_world_features(cvt_path, trgspk, conf, spkr_conf):
+    x, fs = sf.read(str(cvt_path))
+    x = np.array(x, dtype=np.float)
+    x = low_cut_filter(x, fs, cutoff=70)
+    fe = FeatureExtractor(
+        analyzer="world",
+        fs=conf["feature"]["fs"],
+        fftl=conf["feature"]["fftl"],
+        shiftms=conf["feature"]["shiftms"],
+        minf0=spkr_conf[trgspk]["minf0"],
+        maxf0=spkr_conf[trgspk]["maxf0"],
+    )
+    cvt_f0, _, _ = fe.analyze(x)
+    cvt_mcep = fe.mcep(dim=conf["feature"]["mcep_dim"],
+                       alpha=conf["feature"]["mcep_alpha"])
+    return cvt_mcep, cvt_f0
+
+def get_basename(path):
+    return os.path.splitext(os.path.split(path)[-1])[0]
+
+def calculate(file_list, gt_file_list, conf, spkr_conf, MCD):
+    
+    for i, cvt_path in enumerate(file_list):
+        basename = get_basename(cvt_path)
+        number, srcspk, trgspk = basename.split("_")
+        trgspk = trgspk.split("-")[-1]
+        srcspk = srcspk.split("-")[-1]
+        
+        # get converted features. If mcep, from h5; else waveform
+        if conf["feat_type"] == "mcep":
+            cvt_mcep = read_feature(cvt_path, "feat")
+            cvt_f0 = read_feature(cvt_path, "f0")
+        else:
+            cvt_mcep, cvt_f0 = get_world_features(cvt_path, trgspk, conf, spkr_conf)
+
+        # get ground truth features
+        gt_mcep = read_feature(gt_file_list[f"{trgspk}_{number}"], "mcep")
+        gt_f0 = read_feature(gt_file_list[f"{trgspk}_{number}"], "f0")
+
+        # non-silence parts
+        gt_idx = np.where(gt_f0>0)[0]
```

Comment:
I usually use this func by ```npow```. Do you usually use voiced part to calculate mel-cd?

Reply:
> I usually use this func by `npow`. 

What do you mean by using `npow`?

> Do you usually use voiced part to calculate mel-cd?

Yes.

Reply:
```npow``` is power of a frame which is used in sprocket to discard silence part. 
https://github.com/k2kobayashi/sprocket/blob/4f1408ac487e046a502a6ca416a06955d5cb4a71/sprocket/speech/parameterizer.py#L6

I think it's more appropriate to use ```npow``` instead of uv symbols to discard silence part. But it is not necessary and a hassle to implement it.